### DISCUSSION TITLE
stump: Avoid overlap of best neighbours when k > 1

### DISCUSSION THREAD

**Discussion by Ma11heuu** [TDAmeritrade/stumpy#1041] [Q&A]
I am new to stumpy.
When using stump with k>1 (to find more than one best neighbour), is there a way to avoid the overlap of the k neighbours found? The window size m should define the "exclusion zone" for overlapping.

**Comment by seanlaw** [comment]
@Ma11heuu Welcome to the STUMPY community and thank you for your question. 

> When using stump with k>1 (to find more than one best neighbour), is there a way to avoid the overlap of the k neighbours found? The window size m should define the "exclusion zone" for overlapping. 

Unfortunately, there is no obvious way to avoid having the k neighbors overlap. I recall having a similar discussion with @NimaSarajpoor about this and we came to the conclusion that this would be non-trivial/impossible to implement.

**Reply by Ma11heuu** [reply]
Thank you for your quick reply!
